tensorboard/components/vz_line_chart2/microbenchmark/reporter.ts

```diff
@@ -0,0 +1,169 @@
+          avgTimeInGcInMs,
+        };
+      }
+    )
+    .map(
+      ({
+        name,
+        min,
+        max,
+        numIterations,
+        variance,
+        avgTimeInMs,
+        avgTimeInGcInMs,
+      }) => {
+        return {
+          name,
+          numIterations,
+          avgTime: `${avgTimeInMs.toFixed(3)}ms / run`,
+          min: `${min.toFixed(3)}ms`,
+          max: `${max.toFixed(3)}ms`,
+          stdDeviation: Math.sqrt(variance).toFixed(6),
+          avgTimeInGcInMs: avgTimeInGcInMs
+            ? `${avgTimeInGcInMs.toFixed(3)}ms`
+            : 'N/A',
+        };
+      }
+    );
+
+  const reporter = document.createElement('table');
+  Object.assign(reporter.style, {
+    borderSpacing: '10px',
+    borderCollapse: 'separate',
+  });
+
+  function createTHead() {
+    const header = document.createElement('thead');
+    const headerRow = document.createElement('tr');
+    const name = document.createElement('td');
+    name.innerText = 'Name';
+    const iterations = document.createElement('td');
+    iterations.innerText = 'Iterations';
+    const avgTime = document.createElement('td');
+    avgTime.innerText = 'Avg Time';
+    const min = document.createElement('td');
+    min.innerText = 'Min';
+    const max = document.createElement('td');
+    max.innerText = 'Max';
+    const stdDeviation = document.createElement('td');
+    stdDeviation.innerText = 'Std Deviation';
+    const gc = document.createElement('td');
+    gc.innerText = 'Avg GC Time';
+
+    (headerRow as any).append(
+      name,
+      iterations,
+      avgTime,
+      min,
+      max,
+      stdDeviation,
+      gc
+    );
+    header.appendChild(headerRow);
+    return header;
+  }
+
+  // CREATE BODY
+  const reporterContent = displayResults.map(
+    ({
+      avgTime,
+      name,
+      numIterations,
+      min,
+      max,
+      stdDeviation,
+      avgTimeInGcInMs,
+    }) => {
+      const row = document.createElement('tr');
+      const nameEl = document.createElement('td');
+      nameEl.innerText = name;
+      const iterationsEl = document.createElement('td');
+      iterationsEl.innerText = String(numIterations);
+      const avgTimeEl = document.createElement('td');
+      avgTimeEl.innerText = avgTime;
+      const minEl = document.createElement('td');
+      minEl.innerText = min;
+      const maxEl = document.createElement('td');
+      maxEl.innerText = max;
+      const stdDeviationEl = document.createElement('td');
+      stdDeviationEl.innerText = String(stdDeviation);
+      const gc = document.createElement('td');
+      gc.innerText = avgTimeInGcInMs;
+      (row as any).append(
+        nameEl,
+        iterationsEl,
+        avgTimeEl,
+        minEl,
+        maxEl,
+        stdDeviationEl,
+        gc
+      );
+      return row;
+    }
+  );
+
+  (reporter as any).append(createTHead(), ...reporterContent);
+  (document.body as any).prepend(reporter);
+}
+
+export function consoleReporter(results: Result[]) {
```

Comment:
Is this function used anywhere?

Reply:
Not currently but it is exported so we can use it in the future. Let me know if you want it to be removed.

Reply:
I'm fine with keeping it.